src/Rings/mpoly-ideals.jl

```diff
@@ -415,10 +415,16 @@ julia> K, m = saturation_with_index(I)
 function saturation_with_index(I::MPolyIdeal{T}, J::MPolyIdeal{T} = ideal(base_ring(I), gens(base_ring(I)))) where T
   # `remove` is faster than `Singular.saturation` when saturating with
   # respect to principal ideals
-  if ngens(I) == 1 && ngens(J) == 1
-    is_unit(J[1]) && return (I, base_ring(I)(0))
-    pair = remove(I[1], J[1])
-    return (ideal(pair[2]), pair[1])
+  if base_ring(I) isa Field && is_known(is_principal, I) && is_principal(I) && is_known(is_principal, J) && is_principal(J)
+    is_unit(principal_generator(I)) && return (I, base_ring(I)(0))
+
+    # We can use `remove` if its second argument is irreducible.
+    # This is true for example if the polynomial is of degree 1.
+    # This is the most important case for strict transforms in toric blowups.
```

Comment:
Maybe just clarify the comment so that someone else will still understand it in 10 years.
```suggestion
    # cheap heuristic to allow a speedup by the use of 
    # remove insted of Singular's saturation precisely in the following setting:
    #  I, J known to be principal, J known to be irreducible (degree 1 over a field)
    # main use-case: strict transforms in toric blow ups
```